src/exts/js/testing.ts

```diff
@@ -152,209 +161,214 @@ class Assertion {
+      this.value >= expected,
+      `Expected ${this.value} to be greater than or equal to ${expected}`,
+      {
+        expected,
+        actual: this.value,
+      }
+    );
   }
 
   toBeLessThan(expected: any) {
-    if (this.invert ? this.value < expected : this.value >= expected) {
-      throw new AssertionError(
-        `Expected ${this.value} to be less than ${expected}`,
-        {
-          expected,
-          actual: this.value,
-        },
-      );
-    }
+    this.evaluate(
+      this.value < expected,
+      `Expected ${this.value} to be less than ${expected}`,
+      {
+        expected,
+        actual: this.value,
+      }
+    );
   }
 
   toBeLessThanOrEqual(expected: any) {
-    if (this.invert ? this.value <= expected : this.value > expected) {
-      throw new AssertionError(
-        `Expected ${this.value} to be less than or equal to ${expected}`,
-        {
-          expected,
-          actual: this.value,
-        },
-      );
-    }
+    this.evaluate(
+      this.value <= expected,
+      `Expected ${this.value} to be less than or equal to ${expected}`,
+      {
+        expected,
+        actual: this.value,
+      }
+    );
   }
 
   toBeCloseTo(expected: any, delta: number) {
-    if (
-      this.invert
-        ? Math.abs(this.value - expected) <= delta
-        : Math.abs(this.value - expected) > delta
-    ) {
-      throw new AssertionError(
-        `Expected ${this.value} to be close to ${expected}`,
-        {
-          expected,
-          actual: this.value,
-        },
-      );
-    }
+    this.evaluate(
+      Math.abs(this.value - expected) <= delta,
+      `Expected ${this.value} to be close to ${expected}`,
+      {
+        expected,
+        actual: this.value,
+      }
+    );
   }
 
   toHaveLengthOf(expected: any) {
-    if (
-      this.invert
-        ? this.value.length === expected
-        : this.value.length !== expected
-    ) {
-      throw new AssertionError(`Expected array to have length of ${expected}`, {
+    this.evaluate(
+      this.value.length === expected,
+      `Expected array to have length of ${expected}`,
+      {
         expected,
         actual: this.value.length,
-      });
-    }
+      }
+    );
   }
 
   toHavePropertyOf(expected: any) {
-    if (this.invert ? this.value[expected] : !this.value[expected]) {
-      throw new AssertionError(`Expected object to have property ${expected}`, {
+    this.evaluate(
+      this.value[expected],
+      `Expected object to have property ${expected}`,
+      {
         expected,
         actual: this.value,
-      });
-    }
+      }
+    );
   }
 }
 
 function test(description: string, fn: () => void) {
+  const startTime = new Date();
   try {
     fn();
-    console.log(`✅ ${description}`);
+    const endTime = new Date();
+    console.log(
+      `\n✅ [${startTime.toLocaleTimeString()}] ${description} - Passed`
+    );
+    console.log(
+      `   Execution Time: ${endTime.getTime() - startTime.getTime()}ms`
```

Comment:
Instead of Date, I think it'd be better to use `performance.now()` as it provides more accurate timing (wemay need to truncate it though). Other than that, it looks good.